File: 11.1P/Clock.py
class Clock():
    
    def __init__ (self, ):
        self._second = 0
        self._minute = 0
        self._hour = 0

    def Tick(self):

        self._second += 1
        if self._second == 60:
            self._minute += 1
            self._second = 0

            if self._minute == 60:
                self._hour += 1
                self._minute = 0

                if self._hour == 24:
                    self._hour = 0


    def PrintTime(self):
        return str("%02d"%self._hour) + ":" + str("%02d"%self._minute) + ":"  + str("%02d"%self._second) 

#Main Function:

time = Clock()
for i in range(0,3600):
    time.Tick()
    print(time.PrintTime())

File: 3.3P/3.3P/Clock.cs
using System;
using System.Collections.Generic;
using System.Text;
using Unit_Test;

namespace CounterClock
{
    public class Clock
    {
        private Counter _second;
        private Counter _minute;
        private Counter _hour;
        public Clock()
        {

            _second = new Counter("seconds");
            _minute = new Counter("minutes");
            _hour = new Counter("hours");

        }

        public void Tick()
        {
            _second.Increment();
            if (_second.Ticks == 60)
            {
                _minute.Increment();
                _second.Reset();
                if (_minute.Ticks == 60)
                {
                    _hour.Increment();
                    _minute.Reset();    
                }
            }
        }

        public string ReadTime
        {
            get
            {
                return $"{_hour.Ticks:00}:{_minute.Ticks.ToString("00")}:{_second.Ticks.ToString("00")}";
            }
        }

        public void ResetClock()
        {
            _second.Reset();
            _minute.Reset();
            _hour.Reset();
        }
    }
}
